composer/core/state.py

```diff
@@ -956,29 +1006,22 @@ def load_optim_state(self, state_dict: Dict[str, Any]):
                 continue
             optim_state_dict = serialized_value[type(optimizer).__qualname__]
             if self.fsdp_enabled:
+                assert self.fsdp_state_dict_type is not None
```

Comment:
nit: raise `ValueError` or any other Exception type with a descriptive error message instead of `assert` ?

Reply:
this is just for pyright, will add a comment